packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -0,0 +1,51 @@
+import { Event, getEventJson } from './event';
+import { CfnDetectorModel } from './iotevents.generated';
+
+/**
+ * Properties for defining a state of a detector
+ */
+export interface StateProps {
+  /**
+   * The name of the state
+   */
+  readonly stateName: string;
+
+  /**
+   * Specifies the actions that are performed when the state is entered and the `condition` is `TRUE`
```

Comment:
I have to say, I find the constant name changes here super confusing 😕. This is a good example. We are writing documentation for a property called `onEnterEvents`, its type is `Event[]`, and the description says "Specifies the actions that are performed". Doesn't even mention the word "event" anywhere.

Are we missing something here? Should what in this PR is called an `Event` actually be some sort of Action?

Reply:
### Naming
In CloudFormation, [`State` has `OnEnter`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iotevents-detectormodel-state.html#aws-properties-iotevents-detectormodel-state-properties) and [`OnEnter` has only `events` (typed as `Event[]`)](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iotevents-detectormodel-onenter.html#aws-properties-iotevents-detectormodel-onenter-properties). I easy concated `onEnter` and `events`...

If I named it just `onEnter`, would it be less uncomfortable?

### JSDoc
I easy copy [description from CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iotevents-detectormodel-onenter.html#aws-properties-iotevents-detectormodel-onenter-properties)...

And I didn't implement `actions` yet, becase `actions` is optional.
![](https://user-images.githubusercontent.com/11013683/146963656-ed5ead70-a57f-4702-bc9b-2e377bbab490.png)

But as a result,  it became so confusable, so `actions` should be implemented I think. Can I implement it in this PR? (It will make this PR little bigger)

Reply:
I fix it in [`78fc88c` (#18049)](https://github.com/aws/aws-cdk/pull/18049/commits/78fc88c67e0e7d63d6c7ddb98cf50f3f3ba4d02b).